code/script.js

```diff
@@ -1,325 +1,380 @@
+    }
   }
-}
+};
 
-// It'll filter the characters array and redraw the game board.
+//Alerting the user and filtering the characters
 const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
-  if (category === 'accessories') {
+  const { category, value } = currentQuestion;
+  if (category === "color") {
     if (keep) {
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
+        `Yes, the dog has ${value} fur! Keeping everyone with ${value} fur.`
+      );
+      charactersInPlay = charactersInPlay.filter(
+        (dog) => dog[category] === value
+      );
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
+        `No, the dog doesn't have ${value} fur! Removing everyone with ${value} fur.`
+      );
+      charactersInPlay = charactersInPlay.filter(
+        (dog) => dog[category] !== value
+      );
     }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
+  } else if (category === "job") {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(`Yes, the dog works as a ${value}! Keeping all ${value}s.`);
+      charactersInPlay = charactersInPlay.filter(
+        (dog) => dog[category] === value
+      );
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(`No, the dog doesn't work as a ${value}! Removing all ${value}s.`);
+      charactersInPlay = charactersInPlay.filter(
+        (dog) => dog[category] !== value
+      );
+    }
+  } else if (category === "accessories") {
+    if (keep) {
+      alert(
+        `Yes, the dog is wearing ${value}! Keeping everyone wearing ${value}.`
+      );
+      charactersInPlay = charactersInPlay.filter((dog) =>
+        dog[category].includes(value)
+      );
+    } else {
+      alert(
+        `No, the dog doesn't wear ${value}! Removing everyone wearing ${value}.`
+      );
+      charactersInPlay = charactersInPlay.filter(
+        (dog) => !dog[category].includes(value)
+      );
+    }
+  } else if (category === "other") {
+    if (keep) {
+      alert(`Yes, the dog is a ${value}! Keeping all ${value}s.`);
+      charactersInPlay = charactersInPlay.filter((dog) =>
+        dog[category].includes(value)
+      );
+    } else {
+      alert(`No, the dog is not a ${value}! Removing all ${value}s.`);
+      charactersInPlay = charactersInPlay.filter(
+        (dog) => !dog[category].includes(value)
+      );
     }
   }
+  questions--; //Removes 1 guess from the guess counter
+  questionCounter.innerHTML = questions; //Shows game counter to player
+  generateBoard();
+  dropdown.selectedIndex = null; //Clears the dropdown from previously chosen values
```

Comment:
Great idea to clear the dropdown before every question, makes the game look more fluid and well-designed.